modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -222,6 +231,49 @@ public String getTypeDeclaration(Property p) {
     }
 
     @Override
+    public String toDefaultValue(Property p) {
+        if (p instanceof StringProperty) {
```

Comment:
I'm wondering if the code in this method could be cleaner/shorter.

This method treats DoubleProperty, FloatProperty, IntegerProperty (and others) exactly the same. 
I think a better solution might be to do:

```
if (p instanceof Integerproperty 
    || p instanceof FloatProperty
    || p instanceof ....) {
    if (p.getDefault() != null) return p.getDefault().toString()
} else if (p instanceof StringProperty) {
 /* code here */
}
// default if the property doesn't have a proper default (e.g. DateProperty)
return UNDEFINED_VALUE 
```

I'm not set on this, but the current code has a lot of repetitions of the same statements.

Reply:
I would prefer that as well, but `getDefault()` is not an abstract method of `Property`. Actually, this is the way it is handled in the super class `DefaultCodegen` as well.